src/orion/algo/hyperband.py

```diff
@@ -376,23 +432,26 @@ def _refresh_brackets(self):
             if self.executed_times < self.repetitions:
                 self.append_brackets()
 
-    def append_brackets(self):
+    def append_brackets(self) -> None:
+        assert self.brackets is not None
         self.brackets = self.brackets + self.create_brackets()
         # Reset brackets seeds
         self.seed_brackets(self.seed)
 
-    def create_brackets(self):
+    def create_brackets(self) -> list[HyperbandBracket]:
+        assert self.budgets is not None
         return [
             self.create_bracket(i, bracket_budgets, self.executed_times + 1)
             for i, bracket_budgets in enumerate(self.budgets)
         ]
 
-    def _get_bracket(self, trial):
+    def _get_bracket(self, trial: Trial) -> HyperbandBracket:
         """Get the bracket of a trial"""
         _id_wo_fidelity = self.get_id(trial, ignore_fidelity=True, ignore_parent=True)
+        assert self.brackets is not None
```

Comment:
Another

Reply:
Fixed in 35ce6867769d77028c87f7bb015adfd349315f85